modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -627,7 +627,8 @@ def reset_index(self, **kwargs):
         else:
             new_self = self.copy()
             new_self.index = (
-                pandas.RangeIndex(len(new_self.index))
+                # Cheaper to compute row lengths than index
+                pandas.RangeIndex(sum(new_self._modin_frame._row_lengths))
```

Comment:
Do we really want to make QC know that `_modin_frame` has `_row_lengths`? Same question got hung in #4460.

Reply:
cc @mvashishtha, @devin-petersohn, @RehanSD, @vnlitvinov

Reply:
@YarShev Why not? The modin is based on this mechanism, in addition, this information can be used to advantage in some functions. I think that we can make these functions public (`_row_lengths/_column_widths`).

Reply:
I am okay with this ([link](https://github.com/modin-project/modin/pull/4460#discussion_r876251431)) but @devin-petersohn, @mvashishtha might have different thoughts on the matter. I would like to make sure we are on the same page.

Reply:
I support making `row_lengths` and `column_widths` public.

Reply:
Good, let's make those public. Please create an issue for that.

Reply:
https://github.com/modin-project/modin/issues/5024